### PR TITLE
Switch from Bower to npm for front-end dependencies

### PR DESCRIPTION
**Pull request by Nateowami**
This is not ready to be merged, but most of the tests are passing (two are failing in the editor entry section). I intend to squash before merging. This is very much raw work (not even ready to be reviewed), but I'm putting it here so it's easy to see what I'm working on. Will add a note when it's ready.

---
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/web-languageforge/141)

### CODE REVIEW

**Comment by megahirt** [comment]
Any chance we could get the npm bower change over as a separate step from the BS4 alpha 5 work?  It's hard for me to review together

---

Review status: 0 of 19 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/sillsdev/web-languageforge/141#-:-KosA7jd8KH4uyvqBz9u:b9ru95c)*

**Comment by Nateowami** [comment]
Actually there is no BS4 alpha 5 work (we're already on alpha 5). I'm guessing the commit messages were confusing. What happened was, in the first commit, when I installed Bootstrap, I accidentally installed alpha 6. I also installed angular-zxcvbn, but missed zxcvbn itself. So in the next commit I installed zxcvbn and downgraded Bootstrap to alpha 5.

I was planning to squash this and give it a more meaningful message before it actually gets reviewed. But I thought I'd been working on it for awhile, so I should actually push it to a branch, and opened this PR so it would be clear what I was actually working on (branches don't seem very visible). Would it be better if I just leave it as a branch with no PR?

---

Review status: 0 of 19 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/sillsdev/web-languageforge/141#-:-KostvZhOB0WAbdZN6r6:b1ncupj)*

File: refreshDeps.sh [hunk 1]
```diff
@@ -4,7 +4,7 @@ rm -r node_modules
 npm install &
 cd src
 composer install &
-rm -r vendor_bower
+rm -r node_modules
 bower install &
```
Comment (irahopkinson) [inline refreshDeps.sh]: change `bower install &` to `npm install &`

**Comment by irahopkinson** [comment]
For me branches are plenty visible (I use a GUI to see the git tree) however a PR does ensure visibility; more than I need though. PRs might be best when you actually want it merged.

**Comment by Nateowami** [comment]
Good to know. I use the CLI almost exclusively, so I don't really notice branches that I'm not directly involved with. I'll probably just push branches in the future then.

It would probably be a good idea to grep the project for mentions of bower as a final step; I know there's mentions of Bower left and I doubt I will find them all otherwise.

**Comment by Nateowami** [comment]
Everything is now switched to npm. I've yet to remove Bower-related files and update scripts and documentation accordingly (plan to do that Monday), but tests are passing and Bower is no longer being used.

In the src/package.json file many of the dependencies are specified at an exact version rather than a version range. This was in order to preserve dependency versions across the transition so only one thing is changing at a time.
The only dependency version change I am aware of is Open Sans. It was previously at 1.0.0, now it is at 1.4.0. npm does not have an Open Sans package at 1.0.0, and I wonder whether we were actually using 1.0.0, or whether it was just the default version number in bower.json.

**Comment by irahopkinson** [comment]
I don't think anything is using the **Open Sans**. I think I added the dependency but the work to use it was never completed. @megahirt do you know anything about if its being used?

**Comment by irahopkinson** [comment]
This is another good step towards using TypeScript. I would like to see the Ansible deploy and documentation changes added before merging this.

**Comment by megahirt** [comment]
Let's just remove the open sans dependency if possible.  Less is more.  I don't know of any use of that dependency.  Thanks.

I agree that we should have the Ansible and Documentation updated to reflect this change.  Thanks @nateowami 

---

Review status: 0 of 19 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/sillsdev/web-languageforge/141#-:-KpG8Ooyz6CnSE8fUmqJ:b9jkf6p)*

**Comment by Nateowami** [comment]
I believe I've correctly removed/updated all references to Bower. Sass builds properly, but I haven't actually tested the Ansible changes. It would be nice to do that on a fresh machine. I'll run the tests once more when I have a moment.

**Comment by Nateowami** [comment]
E2e tests are passing, Open Sans has been removed, documentation has been updated, and the Ansible setup changed. Should be merge-ready.

**Comment by irahopkinson** [comment]
:lgtm:

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/sillsdev/web-languageforge/141#-:-KpPQ-giqz5H19Chgoho:bnfp4nl)*

**Comment by megahirt** [comment]
Great!  I will squash commits and then close this PR.  :lgtm: 

---

Reviewed 15 of 19 files at r1, 1 of 3 files at r3, 7 of 7 files at r4, 2 of 2 files at r5.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/sillsdev/web-languageforge/141#-:-KpQ09mSyo5Llj3xGpAg:b28psrz)*

**Comment by megahirt** [comment]
This was merged with a `git merge --squash` and resulted in commit 1e8485549a22d530c50de24485f69df15c775cd5

**Pull request closed without merging**